optimade/server/entry_collections/elasticsearch.py

```diff
@@ -184,15 +188,28 @@ def _run_db_query(
 
         search = search.sort(*elastic_sort)
 
-        search = search[page_offset : page_offset + limit]
+        if page_offset:
+            search = search[page_offset : page_offset + limit]
+
+        elif page_above:
+            search = search.extra(search_after=page_above, limit=limit)
+
+        else:
+            search = search[0:limit]
+            page_offset = 0
+
         search = search.extra(track_total_hits=True)
         response = search.execute()
 
         results = [hit.to_dict() for hit in response.hits]
 
         if not single_entry:
             data_returned = response.hits.total.value
-            more_data_available = page_offset + limit < data_returned
+            if page_offset is not None:
+                more_data_available = page_offset + limit < data_returned
+            else:
+                # Needs to be set by some custom elastic query: is the ID the last ID?
+                more_data_available = False
```

Comment:
```suggestion
                # Needs to be set by some custom elastic query: is the ID the last ID?
                if len(results)<limit:
                    more_data_available = False
                elif data_returned == len(results):
                    more_data_available = False
                else:
                     more_data_available = True
```
I think this code will give the correct value with a probability of at least (limit-1)/limit.

Reply:
I've just pushed something to this effect, though tricky to test until we have a proper working implementation for ES `page_above`.